File: 2022/Day16/FastNode.cs
public class FastNode
{



    public string Name;
    public int Flow;
    public int Index;
    public bool IsOpened=false;
    public Dictionary<FastNode,int> Adjenced=new Dictionary<FastNode,int>();

    public FastNode(string name, int flow, int index)
    {
        this.Name = name;
        this.Flow = flow;
        this.Index = index;
    }
    public void AddAdjenced(FastNode node,int wei) { Adjenced.Add(node,wei); }



    public override bool Equals(object? obj)
    {
        return Name.Equals(obj);
    }

    public override int GetHashCode()
    {
        return Name.GetHashCode();
    }
}

File: 2022/Day16++/FastNode.cpp
#pragma once
#include "FastNode.h"
#include "Node.h"
#include "CaveLayout.h"
//
//size_t MyHashFunction::operator()(const FastNode& fn)
//{
//    return (hash<string>()(fn.Name));
//}
//
//
//size_t FastNode::operator()() 
//{
//    return (hash<string>()(Name));
//}
//size_t FastNode::operator()(const FastNode& fn) const 
//{
//    return (hash<string>()(fn.Name))^ (hash<string>()(this->Name));
//}
//
//
//bool FastNode::operator==(const FastNode& lhs) {
//    if (this->Name == lhs.Name)
//        return true;
//    else
//        return false;
//}
//bool FastNode::operator!=(const FastNode& lhs) {
//    if (this->Name == lhs.Name)
//        return false;
//    else
//        return true;
//}

    FastNode::FastNode()
{
        Name = "";
    Flow = 0;
    Index = 0;
}
    FastNode::FastNode(string name, int flow, int index)
{
    Name = name;
    Flow = flow;
    Index = index;
}

void FastNode::AddAdjenced(std::shared_ptr<FastNode> node, int wei) { Adjenced[node->Name]=(wei); }

int FastNode::GetValueToEnd(int timeWhenIsOpened, int timeOfEnd)
{
    int val = (timeOfEnd - timeWhenIsOpened) * Flow;
    return val;
}